sdk/eventgrid/azure-eventgrid/azure/eventgrid/_patch.py

```diff
@@ -7,92 +7,122 @@
-        **kwargs: Any
+        **kwargs: Any,
     ) -> None:
-        _endpoint = "{endpoint}"
-        self._level = level
+        self._namespace = namespace_topic
+        self._credential = credential
 
-        if level == ClientLevel.BASIC:
-            self._client = EventGridPublisherClient(
+        if not self._namespace:
+            self._client = LegacyEventGridPublisherClient(
                 endpoint,
                 credential,
                 api_version=api_version or DEFAULT_BASIC_API_VERSION,
             )  # type:ignore[assignment]
-            self._send = self._client.send  # type:ignore[attr-defined]
-        elif level == ClientLevel.STANDARD:
+            self._publish = self._client.send  # type:ignore[attr-defined]
+        else:
             if isinstance(credential, AzureSasCredential):
                 raise TypeError("SAS token authentication is not supported for the standard client.")
             super().__init__(
                 endpoint=endpoint,
                 credential=credential,
                 api_version=api_version or DEFAULT_STANDARD_API_VERSION,
-                **kwargs
+                **kwargs,
             )
 
-            self._send = self._publish_cloud_events
-        else:
-            raise ValueError("Unknown client level. Known values are `Standard` and `Basic`.")
+            self._publish = self._send_events
         self._serialize = Serializer()
         self._deserialize = Deserializer()
         self._serialize.client_side_validation = False
 
+    def __repr__(self) -> str:
+        return (
+            f"<EventGridPublisherClient: namespace_topic={self._namespace}, credential type={type(self._credential)}>"
+        )
+
+
+class EventGridConsumerClient(InternalEventGridConsumerClient):
+    """EventGridConsumerClient.
+
+    :param endpoint: The host name of the namespace, e.g.
+     namespaceName1.westus-1.eventgrid.azure.net. Required.
+    :type endpoint: str
+    :param credential: Credential used to authenticate requests to the service. Is either a
+     AzureKeyCredential type or a TokenCredential type. Required.
+    :type credential: ~azure.core.credentials.AzureKeyCredential or
+     ~azure.core.credentials.TokenCredential
+    :keyword namespace_topic: The name of the topic to consume events from. Required.
+    :paramtype namespace_topic: str
+    :subscription: The name of the subscription to consume events from. Required.
+    :paramtype subscription: str
+    :keyword api_version: The API version to use for this operation. Default value is "2024-06-01".
+     Note that overriding this default value may result in unsupported behavior.
+    :paramtype api_version: str
+    """
+
+    def __init__(
+        self,
+        endpoint: str,
+        credential: Union[AzureKeyCredential, "TokenCredential"],
+        *,
+        namespace_topic: str,
+        subscription: str,
+        api_version: Optional[str] = None,
```

Comment:
was `api_version` being `None` already GA ? otherwise we can simply make it DEFAULT_STANDARD_API_VERSION and match the docstrings

Reply:
its GA this way in the publisherClient which matches our other libraries that default api_version to None